### PR TITLE
Added downloader for YT-DLP, and partially fixed tests

### PR DESCRIPTION
**Pull request by micah686**
Removed the powershell script which was broken and would download the files into the wrong folders.
Changed the tool to yt-dlp instead of youtube-dl
Added two functions to get platform-dependent yt-dlp and ffmpeg binaries.
Added a github workflows to automatically build and test the tool.

In tests, DownloadTests needs to be modified, because they are failing with this error:
`Assert.AreEqual failed. Expected:<>. Actual:<WARNING: External Downloader arguments given without specifying name. The arguments will be given to all external downloaders>.`

### CODE REVIEW

File: YoutubeDLSharp.Tests/MetadataTests.cs [hunk 1]
```diff
@@ -50,6 +51,7 @@ public async Task TestVideoInformationVimeo()
         [TestMethod]
         public async Task TestPlaylistInformation()
         {
+            System.Diagnostics.Trace.WriteLine(System.IO.Directory.GetCurrentDirectory());
```
Comment (alxnull) [inline YoutubeDLSharp.Tests/MetadataTests.cs:54]:
```suggestion
```

File: YoutubeDLSharp.sln [hunk 2]
```diff
@@ -1,44 +1,37 @@
 ﻿
```
Comment (alxnull) [inline YoutubeDLSharp.sln:1]: Please revert changes in this file, thanks!

File: YoutubeDLSharp/Helpers/DownloadHelper.cs [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+    internal class DownloadHelper
+    {
+        /// <summary>
+        /// Downloads the YT-DLP binary depending on OS
+        /// </summary>
+        /// <param name="directoryPath">The optional directory of where it should be saved to</param>
+        /// <exception cref="Exception"></exception>
+        internal static void DownloadYtDlp(string directoryPath = "")
+        {
+            const string baseGithubPath = "https://github.com/yt-dlp/yt-dlp/releases/latest/download/yt-dlp";
```
Comment (alxnull) [inline YoutubeDLSharp/Helpers/DownloadHelper.cs:26]:
```suggestion
            const string BASE_GITHUB_URL = "https://github.com/yt-dlp/yt-dlp/releases/latest/download/yt-dlp";
```

File: YoutubeDLSharp.Tests/YoutubeDLSharp.Tests.csproj [hunk 4]
```diff
@@ -19,9 +19,6 @@
     <None Update="Lib\ffmpeg.exe">
       <CopyToOutputDirectory>Always</CopyToOutputDirectory>
     </None>
```
Comment (alxnull) [inline YoutubeDLSharp.Tests/YoutubeDLSharp.Tests.csproj:21]:
```suggestion
```

File: .github/workflows/dotnet.yml [hunk 5]
```diff
@@ -0,0 +1,25 @@
+name: .NET
```
Comment (alxnull) [inline .github/workflows/dotnet.yml:1]: This repo is currently using Azure Pipelines for automatic builds & testing. While we can consider switching to GH Actions in the future, please remove this for now, thanks.

File: YoutubeDLSharp.Tests/MetadataTests.cs (hunk 1 above)
Comment (alxnull) [inline YoutubeDLSharp.Tests/MetadataTests.cs:54]:
```suggestion
```

**Review by alxnull — COMMENTED** [review]
Hey @micah686, thanks a lot for opening this PR!

The new binary downloading feature you implemented looks good to me and I'd be happy to merge that. I left some comments at a few locations, would be great if you could consider those. Thanks!

Also, I already fixed some issues with the tests, including those you mentioned, in the latest commit to master, so the test suite [should pass now](https://github.com/Bluegrams/YoutubeDLSharp/runs/8170471329).

**Comment by micah686** [comment]
Ok, I updated the project. The test failure seems to be related to the old `RestoreBinaries.ps1` file, which was removed in the current version.

**Review by alxnull — APPROVED** [review]
Thanks!

**Pull request merged**